File: templategenerator.systemtest/src/test/java/com/optivem/atddaccelerator/templategenerator/systemtest/util/ProcessResultAssertions.java
package com.optivem.atddaccelerator.templategenerator.systemtest.util;

import static org.assertj.core.api.Assertions.assertThat;

public class ProcessResultAssertions {

    public static void assertSuccess(ProcessResult result) {
        assertThat(result.isSuccess()).withFailMessage(() -> "Process finished with exit code " + result.getExitCode() + "\nErrors: " + result.getErrors()).isTrue();
    }

    public static void assertFailure(ProcessResult result) {
        assertThat(result.isError()).withFailMessage(() -> "Process finished with exit code " + result.getExitCode() + "\nErrors: " + result.getErrors()).isTrue();
    }

}

File: console/tests/Util/Process/ProcessResultAssertions.cs
using FluentAssertions;
using Xunit;

namespace Optivem.AtddAccelerator.TemplateGenerator.SystemTests.Util.Process
{
    public static class ProcessResultAssertions
    {
        public static void ShouldSucceed(this ProcessResult result)
        {
            result.IsSuccess.Should().BeTrue(Stringify(result));
        }

        public static void ShouldSucceed(this ProcessResult result, string customMessage)
        {
            result.IsSuccess.Should().BeTrue(Stringify(result, customMessage));
        }

        public static void ShouldFail(this ProcessResult result)
        {
            result.IsError.Should().BeTrue(Stringify(result));
        }

        public static void ShouldFail(this ProcessResult result, string customMessage)
        {
            result.IsError.Should().BeTrue(Stringify(result, customMessage));
        }

        private static string Stringify(ProcessResult result)
        {
            return $"Process finished with exit code {result.ExitCode}\nErrors: {result.Errors}\nOutput: {result.Output}";
        }

        private static string Stringify(ProcessResult result, string customMessage)
        {
            return $"{customMessage}\n{Stringify(result)}";
        }
    }
}